src/AutoRest.CSharp/Common/Output/Models/Types/TypeProvider.cs

```diff
@@ -19,7 +19,9 @@ protected TypeProvider(BuildContext context)
             _existingType = new Lazy<INamedTypeSymbol?>(() => Context.SourceInputModel?.FindForType(DefaultNamespace, DefaultName));
         }
 
-        public CSharpType Type => new(this, TypeKind is TypeKind.Struct or TypeKind.Enum);
+        protected virtual CSharpType[]? Arguments { get; } = null;
```

Comment:
`TypeProvider` is too general type to add `Arguments` property. Let's make `public CSharpType Type` property `virtual` and add arguments in `MgmtLongRunningOperation` directly.